### DISCUSSION TITLE
if let store = self.store.scope(state: \.child, action: \.child) -> Store<Child.State, Child.State?>

### DISCUSSION THREAD

**Discussion by Jaeki-Lee** [pointfreeco/swift-composable-architecture#2916] [Q&A]
I appreciate for you reading my question.

I made like this

```
@Reducer
public struct Parent {
  public init() {}

  @ObservableState
  public struct State: Equatable {
    public init() {}
    var child: Child.State?
  }

  public enum Action: Equatable {
    case child(Child.Action)
  }

  public var body: some ReducerOf<Self> {
    Reduce { state, action in
      switch action {

       case let .child(action):
        switch action {
        case .test:
          print("test")
          return .none
        default: return .none
        }
        
      default: return .none
      }
    }
  }
}
@Reducer
public struct Child {
  public init() {}

  @ObservableState
  public struct State: Equatable {
    public init() {}
  }

  public enum Action: Equatable {
    case test
  }

  public var body: some ReducerOf<Self> {
    Reduce { state, action in
      switch action {
      default: return .none
      }
    }
  }
}
```

```
public struct ParentView {
  @Perception.Bindable
  var store: StoreOf<Parent>

  public init(store: StoreOf<Parent>) {
    self.store = store
  }
}

extension ParentView: View {
  public var body: some View {
    WithPerceptionTracking {
        VStack(content: {

          //The return value Store<Child.State, Child.State?> not Store<Child.State, Child.Action>
          if let store = self.store.scope(state: \.child, action: \.child) {
            ChildView(store: store)
          }

        })
}
```

<img width="740" alt="스크린샷 2024-03-12 오후 6 32 38" src="https://github.com/pointfreeco/swift-composable-architecture/assets/67038343/19080e0a-5860-47e7-9404-7964f723f4a7">

I don't understand why the return of scope(state:, action) is Store<Avatar.State, Avatar.State?

**Comment by Jaeki-Lee** [comment]
if I change the name action name `child` to `childAction` then it seems works

**Reply by mbrandonw** [reply]
Hi @Jaeki-Lee, the code you provided above compiles just fine for me. Are you sure there isn't something else happening in your real code?

**Comment by Jaeki-Lee** [comment]
Thank you for answering.
Yes the picture show the real situation, so I rename avatar action case to "avatarAction" for this issue.

**Reply by mbrandonw** [reply]
Are you able to provide a reproduction in a separate project though? Because the code you provided above does not show the behavior you are describing. It works perfectly fine.

**Reply by Jaeki-Lee** [reply]
I appreciate for your Interesting look.
It works after delete Equatable from child Reducer and State.

**Reply by mbrandonw** [reply]
Hi @Jaeki-Lee, can you please provide a complete code snippet that demonstrates the problem? The code you have provided compiles just fine, with or without the `Equatable` conformances.

**Reply by Jaeki-Lee** [reply]
Yes I am using TCA "1.9.0".
I can't share the code that I am using, but when I got time sure I will make the sample project and share with you. I appreciate for thinking with me cheers!

**Reply by Jaeki-Lee** [reply]
I make sample code for it, but it works as you told me..
However, when the issue happen 
the \.child was state even I put on action parameter.
`let store = self.store.scope(state: \.child, action: \.child)`
